### PR TITLE
[DOC] Declare range API-stability (container/range)

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

**Review by simonsasse — APPROVED** [review]
Looks good :)

File: include/seqan3/range/concept.hpp [hunk 1]
```diff
@@ -37,6 +37,8 @@ namespace seqan3
  *   * certain views store a state with their algorithm that also changes when `begin()` is called or an
  * iterator is incremented; these may be not be `const`-iterable, because the standard library
  * (and also SeqAn) guarantees that it is safe to call `const`-qualified functions concurrently.
+ *
+ * \noapi{Exposition only}
```
Comment (eseiler) [inline include/seqan3/range/concept.hpp:41]:
```suggestion
 * \noapi{Exposition only.}
```

File: include/seqan3/range/concept.hpp [hunk 2]
```diff
@@ -77,6 +79,8 @@ SEQAN3_CONCEPT const_iterable_range =
  *
  * The requirements for this concept are given as related functions and type traits.
  * Types that model this concept are shown as "implementing this interface".
+ *
+ * \noapi{Exposition only}
```
Comment (eseiler) [inline include/seqan3/range/concept.hpp:83]:
```suggestion
 * \noapi{Exposition only.}
```

File: include/seqan3/range/concept.hpp [hunk 3]
```diff
@@ -112,6 +116,8 @@ SEQAN3_CONCEPT pseudo_random_access_iterator =
...
```
Comment (eseiler) [inline include/seqan3/range/concept.hpp:120]:
```suggestion
 * \noapi{Exposition only.}
```

**Review by eseiler — APPROVED** [review]

**Pull request merged**